functions.php

```diff
@@ -91,17 +104,30 @@ function as_schedule_single_action( $timestamp, $hook, $args = array(), $group =
 		return is_int( $pre ) ? $pre : 0;
 	}
 
-	return ActionScheduler::factory()->create(
-		array(
-			'type'      => 'single',
-			'hook'      => $hook,
-			'arguments' => $args,
-			'when'      => $timestamp,
-			'group'     => $group,
-			'unique'    => $unique,
-			'priority'  => $priority,
-		)
-	);
+	try {
+		return ActionScheduler::factory()->create(
```

Comment:
I don't think there's a right or wrong answer here, but what are your thoughts on any of the following permutations:

- Moving the try/catch structures, complete with logging code, inside `ActionScheduler::factory()->create()`?
- Or, moving the logging code inside `ActionScheduler::factory()->create()` but still catch the exception here?

Primarily thinking this would allow us to reduce the degree of repetition across these function definitions.

Reply:
> If we move only the logging code there but not the exception catching we'd not yet have the exception details available ...? (Not sure if I understand correctly?)

My description was a little vague, sorry. *If* we did this (and I was just noting it as a possibility, I much prefer the earlier option) we'd need to catch the exception first, to get the error details, but then re-throw it.

Reply:
@barryhughes I made the changes we discussed, so this is ready for another review. 👍